kernel/src/main/java/org/kframework/main/Main.java

```diff
@@ -123,14 +116,12 @@ public int runApplication() {
         KExceptionManager kem = this.kem.get();
         kem.installForUncaughtExceptions();
         try {
-            int retval = frontEnd.get().main();
-            return retval;
+            return frontEnd.get().main();
         } catch (ProvisionException e) {
             for (Message m : e.getErrorMessages()) {
-                if (!(m.getCause() instanceof KEMException)) {
+                if (!(m.getCause() instanceof KEMException ex)) {
                     throw e;
                 } else {
-                    KEMException ex = (KEMException) m.getCause();
```

Comment:
This works in a negated check and an else block?

Reply:
Remarkably, yes! https://blogs.oracle.com/javamagazine/post/java-pattern-matching-instanceof-flow-scoping